integration-tests/utils.py

```diff
@@ -278,33 +278,63 @@ def asyncify(f):
         except JujuError as e:
             if "already running charm" not in str(e):
                 raise
-    await wait_for_ready(model)
 
+    # When we upgrade from a non-rolling master to a rolling upgrade master
+    # the master units will block waiting for the user to call the upgrade action.
```

Comment:
Ouch. I think we probably do not want this behavior in the charm - it's forcing the user to walk through a manual upgrade that does nothing but clear charm state.

Reply:
The cause of this behaviour is the call to the  any_file_changed method https://github.com/kubernetes/kubernetes/blob/master/cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py#L109 The first call any_file_changed will signal file changes.

Reply:
Yes this does not look good. I wonder what would happen on the initial deployment of the charm. Would we be blocked "Needing" to upgrade?

Reply:
The first setup is doing a "forced" installation of snaps so we are do not block in that case.

On every installation we should be storing the "fingerprint" (aka call any_file_changed without waiting) of the snap resources we are installing so the check any_file_changed is doing is correct even the first time. For the upgrade from non-rolling to rolling master would it be ok not to block waiting for the admin to trigger the upgrade action, @Cynerva ?

Reply:
> For the upgrade from non-rolling to rolling master would it be ok not to block waiting for the admin to trigger the upgrade action, @Cynerva ?

Yeah, that should be fine. Seems worth it to provide a more consistent user experience: `juju upgrade-charm` does not result in a block for manual upgrade. Only changing the channel config or resources should result in that.